src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1087,6 +1089,98 @@ def aks_get_credentials(cmd, client, resource_group_name, name, admin=False,
 }
 
 
+# pylint: disable=line-too-long
+def aks_kollect(cmd, client, resource_group_name, name, storage_account_name=None, sas_token=None, storage_account_id=None,):
+    mc = client.get(resource_group_name, name)
+
+    if not which('kubectl'):
+        raise CLIError('Can not find kubectl executable in PATH')
+
+    if (sas_token is None) != (storage_account_name is None):
+        raise CLIError("Please specify both storage account name and a SAS token with write permission.")
+
+    readonly_sas_token = None
+    if storage_account_name is None:
+        if storage_account_id is None:
+            print("No storage account resource id specified. Try getting storage account from diag settings")
+            storage_account_id = get_storage_account_from_diag_settings(cmd.cli_ctx, resource_group_name, name)
+            if storage_account_id is None:
+                raise CLIError("A storage account must be specified, since there isn't one in the diag settings.")
+
+        from msrestazure.tools import is_valid_resource_id, parse_resource_id
+        if is_valid_resource_id(storage_account_id):
+            try:
+                parsed_storage_account = parse_resource_id(storage_account_id)
+            except CloudError as ex:
+                raise CLIError(ex.message)
+        else:
+            raise CLIError("Invalid storage account id %s" % storage_account_id)
+
+        storage_account_name = parsed_storage_account['name']
+        storage_client = cf_storage(cmd.cli_ctx, parsed_storage_account['subscription'])
+        storage_account_keys = storage_client.storage_accounts.list_keys(parsed_storage_account['resource_group'], storage_account_name)
+        kwargs = {
+            'account_name': storage_account_name,
+            'account_key': storage_account_keys.keys[0].value
+        }
+        cloud_storage_client = cloud_storage_account_service_factory(cmd.cli_ctx, kwargs)
+
+        sas_token = cloud_storage_client.generate_shared_access_signature(
+            'b',
+            'sco',
+            'rwdlacup',
+            datetime.datetime.utcnow() + datetime.timedelta(days=1))
+
+        readonly_sas_token = cloud_storage_client.generate_shared_access_signature(
+            'b',
+            'sco',
+            'rl',
+            datetime.datetime.utcnow() + datetime.timedelta(days=1))
+
+        readonly_sas_token = readonly_sas_token.strip('?')
+
+    print("Confirmed. Diagnostic info will be stored in the storage account %s as outlined in aka.ms/AKSPeriscope." % storage_account_name)
+
+    from knack.prompting import prompt_y_n
+    msg = 'This will deploy a daemon set to your cluster to collect logs and diagnostic information and ' \
+        'save them to the specified storage account as outlined in aka.ms/AKSPeriscope. If you share access ' \
```

Comment:
I checked this short link and it does not work yet.

Reply:
Jennifer will update the link outside of this PR.